lib/logmatcher.c

```diff
@@ -291,8 +291,22 @@ typedef struct _LogMatcherPcreRe
   pcre *pattern;
   pcre_extra *extra;
   gint match_options;
+  gint *matches;
+  gsize matches_size;
 } LogMatcherPcreRe;
 
+static void
+_alloc_matches(LogMatcherPcreRe *self)
+{
+  gint num_matches;
+  if (pcre_fullinfo(self->pattern, self->extra, PCRE_INFO_CAPTURECOUNT, &num_matches) < 0)
+    g_assert_not_reached();
+  if (num_matches > RE_MAX_MATCHES)
+    num_matches = RE_MAX_MATCHES;
```

Comment:
Please add a warning that syslog-ng will only index the first `RE_MAX_MATCHES`.

Reply:
Done, thanks! :)